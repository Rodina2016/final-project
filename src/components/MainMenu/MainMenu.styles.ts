import { createStyles, Theme } from '@material-ui/core/styles'

export default (theme: Theme) => createStyles({
    root: {
        padding: theme.spacing(2, 0),
        marginBottom: 20,
        backgroundColor: theme.palette.primary.main
    },
    wrap: {
        display: 'flex',
        justifyContent: 'flex-end'
    },
    item: {
        marginRight: 20
    },
    link: {
        fontSize: 20,
        color: '#fff',
        textDecoration: 'none',

        '&:hover': {
            opacity: .7,
        }
    }
})