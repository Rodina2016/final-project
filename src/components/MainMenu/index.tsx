import { NavLink, Router } from 'react-router-dom';
import makeStyles from '@material-ui/core/styles/makeStyles';
import styles from './MainMenu.styles';
import React from 'react';
import Container from '@material-ui/core/Container/Container';
import { MainMenuProps } from '../../types';
import history from '../../history';

const useStyles = makeStyles(styles, {
    name: 'MainMenu'
});

const MainMenu: React.FC<MainMenuProps> = (props): JSX.Element => {
    const classes = useStyles(props);

    return (
        <Router history={history}>
            <div className={classes.root}>
                <Container maxWidth={'md'}>
                    <div className={classes.wrap}>
                        <div className={classes.item}>
                            <NavLink to="/" className={classes.link}>
                                Main
                            </NavLink>
                        </div>
                        <div>
                            <NavLink to="/catched-pokemons" className={classes.link}>
                                Catched pokemons
                            </NavLink>
                        </div>
                    </div>
                </Container>
            </div>
        </Router>
    );
};

export default MainMenu;