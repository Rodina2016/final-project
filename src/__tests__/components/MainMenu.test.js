import React from 'react';
import MainMenu from '../../components/MainMenu';
import 'regenerator-runtime/runtime';
import renderer from 'react-test-renderer';

test("main menu snapShot", () => {
    const component = renderer.create(
        <MainMenu/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

})