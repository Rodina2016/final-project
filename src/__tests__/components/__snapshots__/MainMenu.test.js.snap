// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main menu snapShot 1`] = `
<div
  className="MainMenu-root-1"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div
      className="MainMenu-wrap-2"
    >
      <div
        className="MainMenu-item-3"
      >
        <a
          aria-current="page"
          className="MainMenu-link-4 active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Main
        </a>
      </div>
      <div>
        <a
          aria-current={null}
          className="MainMenu-link-4"
          href="/catched-pokemons"
          onClick={[Function]}
        >
          Catched pokemons
        </a>
      </div>
    </div>
  </div>
</div>
`;
